### DISCUSSION TITLE
BT curve and ROR lag

### DISCUSSION THREAD

**Discussion by bolza4** [artisan-roaster-scope/artisan#908] [General]
Hello everyone,

I'm having a hard time tracking, at the same time, the BT curve and the ROR curve on the background. 

It seems like I can only align my current roast with the BT curve on the background OR the ROR curve, in other words I have to choose if I wanna track one curve or the other.
If I track the BT curve (graph 2), I have a delay on the ROR curve, or the other way around as shown in graph 1 where I can easily track ROR but I have issues in tracking the BT curve.

How can I fix it?
thank you so much :)

matteo 

![175905084-6f3a8431-be3d-44c9-a442-89ef186f4b8f](https://user-images.githubusercontent.com/108266430/175907500-7c50ab86-8491-4e7f-b317-bd811bffbe35.png)

![175905165-9ff9547d-53ef-49b7-adb9-2be98bb7f9e4](https://user-images.githubusercontent.com/108266430/175907498-0fbbf8ce-c27a-476f-81a0-743a0f0af3cd.png)

**Comment by poundy** [comment]
can I ask why you care about the ET curve? Remembering that the ET is just a measure of the environment that's helping you get a particular roasting profile for your coffee... and also remembering that it's about the coffee taste at the end of the roast, not how close you stuck one line to another.

We know nothing about your roaster, and nothing about how your between-batch-protocol looks, so we're in the dark here. 

But when I look at the second graph in particular, it is clear that the charge conditions are more than a little different - the TP was a whole 9 seconds different, and the rate of change from charge was noticeably different. I'd also suggest that you have align-on-TP enabled, and I use align-on-charge, as I find that much clearer to indicate where my process has set me up poorly to follow a previous roast.

**Reply by bolza4** [reply]
Hello Poundy,

thanks for your reply :) 

these two graphs belong to the same batch, but aligned differently, and they are just used to get an example of the issue I have. Maybe I didn't express myself clearly due to the fact that english is not my first language, I apologize. the issue I have is that if I move the background in order to match my actual BT curve to the one in the background, my ROR show a delay of 10 seconds compared to the background as shown in the second graph. the same way, if I try to match the ROR on the background to my actual ROR by moving the background, my actual BT curve and the background BT curve have a delay of 10 seconds as shown in the first graph. This makes tracking my previous batch pretty hard.
any idea about how to fix it? 

thanks a lot, appreciate your feedback :)

**Reply by poundy** [reply]
I did not realise we were looking at the same graph just with different alignments, so thanks for clarifying.  I note you've also edited your original post to help clarify that you are concerned about BT and BT-ROR, not ET as it originally mentioned.
Are you comfortable sharing with us the two roast profiles, the background one and the foreground one? That might help speed up discussion about what you're seeing... 
Artisan doesn't have any specific "lag" built into the system. The data is the data, and a ROR is inherently a "lagged" value because it's a rate measure that's calculated based on previous data.  
As I mention in my first reply, the current roast and the background roast "trajectories" are visibly different in the start of the roast, so I can completely understand why ROR will not look the same. My initial reaction therefore is that your assertion about a "delay" is wrong, it's just that things have happened differently in the two roasts, and again the data is the data.... I also think that besides the initial trajectory, the post-TP portion of your graph looks pretty well spot on - I just think your expectation of having the graphs align isn't plausible given the difference in roast...
